File: src/main/java/com/gmail/borlandlp/minigamesdtools/arena/team/lobby/respawn/ExampleRespawnLobbyListener.java
package com.gmail.borlandlp.minigamesdtools.arena.team.lobby.respawn;

import com.gmail.borlandlp.minigamesdtools.arena.ArenaEventListener;
import com.gmail.borlandlp.minigamesdtools.arena.localevent.*;

public class ExampleRespawnLobbyListener implements ArenaEventListener {
    private RespawnLobby lobby;

    public ExampleRespawnLobbyListener(RespawnLobby l) {
        this.lobby = l;
    }

    @ArenaEventHandler(
            ignoreCancelled = true,
            priority = ArenaEventPriority.HIGH
    )
    public void onPlayerDamage(ArenaPlayerDamagedLocalEvent event) {
        if(this.lobby.getWaitingPlayers().containsKey(event.getPlayer())) {
            event.setCancelled(true);
        }
    }

    @ArenaEventHandler(
            ignoreCancelled = true,
            priority = ArenaEventPriority.HIGH
    )
    public void onPlayerKilled(ArenaPlayerDeathLocalEvent event) {
        if(this.lobby.getWaitingPlayers().containsKey(event.getPlayer())) {
            event.setCancelled(true);
        }
    }

    @ArenaEventHandler(
            ignoreCancelled = true,
            priority = ArenaEventPriority.HIGH
    )
    public void onPlayerKilled(ArenaPlayerKilledLocalEvent event) {
        if(this.lobby.getWaitingPlayers().containsKey(event.getPlayer())) {
            event.setCancelled(true);
        }
    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/arena/team/lobby/respawn/ExampleRespawnLobbyListener.kt
package com.gmail.borlandlp.minigamesdtools.arena.team.lobby.respawn

import com.gmail.borlandlp.minigamesdtools.arena.ArenaEventListener
import com.gmail.borlandlp.minigamesdtools.arena.localevent.*

class ExampleRespawnLobbyListener(private val lobby: RespawnLobby) : ArenaEventListener {
    @ArenaEventHandler(ignoreCancelled = true, priority = ArenaEventPriority.HIGH)
    fun onPlayerDamage(event: ArenaPlayerDamagedLocalEvent) {
        if (lobby.waitingPlayers.containsKey(event.player)) {
            event.isCancelled = true
        }
    }

    @ArenaEventHandler(ignoreCancelled = true, priority = ArenaEventPriority.HIGH)
    fun onPlayerKilled(event: ArenaPlayerDeathLocalEvent) {
        if (lobby.waitingPlayers.containsKey(event.player)) {
            event.isCancelled = true
        }
    }

    @ArenaEventHandler(ignoreCancelled = true, priority = ArenaEventPriority.HIGH)
    fun onPlayerKilled(event: ArenaPlayerKilledLocalEvent) {
        if (lobby.waitingPlayers.containsKey(event.player)) {
            event.isCancelled = true
        }
    }

}